src/common/vector/value_vector_utils.cpp

```diff
@@ -26,23 +28,129 @@ void ValueVectorUtils::copyNonNullDataWithSameTypeOutFromPos(const ValueVector&
             copyNonNullDataWithSameTypeOutFromPos(*childVector, pos, dstData, dstOverflowBuffer);
             dstData += childVector->getNumBytesPerValue();
         }
+    } else if (srcVector.dataType.typeID == VAR_LIST) {
+        auto kuList = ValueVectorUtils::convertListEntryToKuList(srcVector, pos, dstOverflowBuffer);
+        memcpy(dstData, &kuList, sizeof(kuList));
     } else {
         copyNonNullDataWithSameType(srcVector.dataType,
             srcVector.getData() + pos * srcVector.getNumBytesPerValue(), dstData,
             dstOverflowBuffer);
     }
 }
 
+void ValueVectorUtils::appendElementToList(
```

Comment:
move to list creation (as a static function in cpp) if no other function need it.